File: String Compression.py
class Solution(object):
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        result = []
        count = 1
        for i in range(1,len(chars)):
            if chars[i] == chars[i-1]:
                count += 1
            else:
                result.append([chars[i-1],count])
                count = 1
        result.append([chars[-1], count])
        i = 0
        for char, count in result:
            chars[i] = char
            i += 1
            if count > 1:
                for item in str(count):
                    chars[i] = str(item)
                    i += 1
        return i

File: Array and String/string-compression/String Compression.go
func compress(chars []byte) int {
    count, index := 1, 0
    for i := 1; i <= len(chars); i++ {
        if i < len(chars) && chars[i] == chars[i - 1] {
            count++
            continue
        }

        chars[index] = chars[i - 1]
        index++
        if count > 1 {
            for _, num := range strconv.Itoa(count) {
                chars[index] = byte(num)
                index++
            }
        }
        count = 1
    }

    return index
}